go/store/types/codec.go

```diff
@@ -368,19 +367,42 @@ func (b *binaryNomsWriter) reset() {
 	b.offset = 0
 }
 
+const (
+	GigsHalf = 1 << 29
+	Gigs2    = 1 << 31
+)
+
 func (b *binaryNomsWriter) ensureCapacity(n uint32) {
-	length := uint32(len(b.buff))
-	if b.offset+n <= length {
+	length := uint64(len(b.buff))
+	minLength := uint64(b.offset + n)
+	if length >= minLength {
 		return
 	}
 
 	old := b.buff
 
-	for b.offset+n > length {
+	if minLength > math.MaxUint32 {
+		panic("overflow")
+	}
+
+	for minLength > length {
```

Comment:
These loops seem like a strange way to ensure that length is bigger than minLength up to MaxUint32. Can't we just round up to the nearest half gig that's over the requested capacity?

Reply:
the bit twiddling approach to rounding to the next power of 2 isn't readable, and this isn't happening in a tight loop where this needs to be super fast so leaving the doubling loop.

If you a good way to round to the next half gig that works, feel free to suggest it.  But I don't think it's more readable, and perf isn't a concern.